LoopProjectFile/ExtractedInformation.py

```diff
@@ -302,72 +313,151 @@ def SetStratigraphicLog(root, data, append=False, verbose=False):
             zlib=True,
             complevel=9,
         )
+        
+        # Define compound types for thickness calculator and active flags (list of length 5)
+        thicknessCalculatorType_t = siGroup.createCompoundType(
+            LoopProjectFile.thicknessCalculatorType, "ThicknessCalculator"
+        )
+        siGroup.createVariable(
+            "thicknessCalculator",
+            thicknessCalculatorType_t,
+            zlib=True,
+            complevel=9,
+        )
+
+        thicknessCalculatorActiveFlagsType_t = siGroup.createCompoundType(
+            LoopProjectFile.thicknessCalculatorType, "ThicknessCalculatorActiveFlags"
+        )
+        siGroup.createVariable(
+            "thicknessCalculatorActiveFlags",
+            thicknessCalculatorActiveFlagsType_t,
+            zlib=True,
+            complevel=9,
+        )
     else:
         siGroup = resp["value"]
 
     if siGroup:
         stratigraphicLayersLocation = siGroup.variables["stratigraphicLayers"]
+        thickness_calculator = siGroup.variables["thicknessCalculator"]
+        thickness_calculator_active_flags_var = siGroup.variables["thicknessCalculatorActiveFlags"]
+        
         index = 0
         if append:
             index = siGroup.dimensions["index"].size
         for i in data:
             stratigraphicLayersLocation[index] = i
             index += 1
         siGroup.setncattr("index_MaxValid", index)
+
```

Comment:
Would it be better to have a single table with the columns [name, active_flag]? So instead of passing the active thickness calculator to the setter, pass the compound type?